src/main/java/frc/robot/subsystems/drive/CommandSwerveDrivetrain.java

```diff
@@ -173,12 +184,29 @@ private void controlDrivetrain() {
         }
     }
 
-    private static Rotation2d calculateDesiredHeading(Pose2d current, Pose2d target) {
+    private Rotation2d calculateDesiredHeading(Pose2d current, Pose2d target) {
         // I hope this is all inlined
         Pose2d robotToTarget = GeomUtil.transformToPose(current.minus(target));
 
+        double distanceToTarget =
+                Math.hypot(
+                        robotToTarget.getTranslation().getX(),
+                        robotToTarget.getTranslation().getY());
+        Translation2d robotToTargetAnticipated =
+                new Translation2d(
+                        robotToTarget.getX()
+                                + (getRobotVelocity.get().getX()
+                                        * noteTimeToGoal.getValue(distanceToTarget)
+                                        * DriveConstants.velocityScalar),
+                        robotToTarget.getY()
+                                + (getRobotVelocity.get().getY()
+                                        * noteTimeToGoal.getValue(distanceToTarget)
+                                        * DriveConstants.velocityScalar));
```

Comment:
The robot's motion shouldn't be based on the note's time to target - once the note is launched we don't care where the robot is. Rather, we care where the robot will need to be when we launch the note, so we need our anticipation time to be some constant to obtain the next robot target pose

Reply:
That is what the velocity scalar would be for. I also incorporated accounting for note velocity here. I guess I could make it add rather than scale?